cos-fleetshard-sync/src/main/java/org/bf2/cos/fleetshard/sync/resources/ConnectorStatusUpdater.java

```diff
@@ -43,4 +81,23 @@ public void update(ManagedConnector connector) {
         }
     }
 
+    /* Expose a metric "cos_fleetshard_sync_connector_state" which reveals the current connector state.
+    Metric value of 1 implies that the connector is in Ready state. Similarly, 2 -> Failed, 3 -> Deleted,
+    4 -> Stopped, 5 -> In Process */
+    private void measure(ManagedConnector connector, Integer connectorState) {
+
+        tags = List.of(
+                Tag.of("cos.connector.id", connector.getSpec().getConnectorId()),
+                Tag.of("cos.deployment.id", connector.getSpec().getDeploymentId()));
+
+        Gauge gauge = registry.find(config.metrics().baseName() + "." + CONNECTOR_STATE).tags(tags).gauge();
+        if (gauge != null) {
+            registry.remove(gauge);
+        }
+
+        Gauge.builder(config.metrics().baseName() + "." + CONNECTOR_STATE, new AtomicInteger(connectorState), AtomicInteger::get)
```

Comment:
This may be simplified by using `builder(String name, Supplier<Number> f)` where the supplied can be cached